packages/unified-pipeline/src/utils/stepUtils/stepIconUtil.ts

```diff
@@ -0,0 +1,27 @@
+import { IconProps } from '@harnessio/canary'
+import {
+  getIsActionStep,
+  getIsBackgroundStep,
+  getIsRunStep,
+  getIsRunTestStep,
+  getIstemplateStep
+} from './commonStepUtil'
+
+export const getIconBasedOnStep = (step: any): IconProps['name'] => {
+  if (getIsRunStep(step)) return 'run'
```

Comment:
It is yet to be refactored by creating a hash map of steps and related icons. So that it can be used across the app.